### PR TITLE
pawel/OPS-5755-round-to-tens

### PR DESCRIPTION
**Pull request by pawelpanasewicz**

### CODE REVIEW

File: app/ssttpcalculator/BigDecimalUtil.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ssttpcalculator
+
+object BigDecimalUtil {
+
+  def roundDownToNearestTen(value: BigDecimal): BigDecimal = BigDecimal((value.longValue() / 10) * 10)
```
Comment (johnrevill) [inline app/ssttpcalculator/BigDecimalUtil.scala]: Iirc from ancient Java days, `BigDecimal` has methods to support careful/exact rounding of numbers avoiding the need for using the long value there. 
Rarely for Java, it's actually an immutable data type on to which you can chain multiple maths operations in sequence.

Comment (johnrevill) [inline app/ssttpcalculator/BigDecimalUtil.scala:21]: Nb..
The BigDecimal.longValue can be a truncating operation - the docs at https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#longValue-- state "any fractional part of this BigDecimal will be discarded"

Comment (johnrevill) [reply app/ssttpcalculator/BigDecimalUtil.scala]: Nb; The `value.longValue` will truncate any fractional part - is this really what's intended? 
I can't tell exactly how it's meant to behave from the `roundUpToNearestTen` function name alone.

File: test/pagespecs/PaymentSummaryPageSpec.scala [hunk 2]
```diff
@@ -59,6 +59,6 @@ class PaymentSummaryPageSpec extends ItSpec {
     paymentSummaryPage.assertPageIsDisplayed
 
     paymentSummaryPage.clickContinue()
-    monthlyPaymentAmountPage.assertPageIsDisplayedAltPath(-100: Int)
+    monthlyPaymentAmountPage.assertPageIsDisplayedAltPath(400, 2390)
```
Comment (johnrevill) [inline test/pagespecs/PaymentSummaryPageSpec.scala]: pedantic: numbers can't be understood without looking at definition  of `assertPageIsDisplayedAltPath`

**Review by johnrevill — APPROVED** [review]

File: test/ssttpcalculator/BigDecimalUtilTest.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      BigDecimalUtil.roundDownToNearestTen(tc.bigDecimal) shouldBe tc.roundedDown
+      BigDecimalUtil.roundUpToNearestTen(tc.bigDecimal) shouldBe tc.roundedUp
+    }
+  }
+
+  lazy val testCases: List[TestCase] = List(
+    TestCase(bigDecimal  = "0.000000001", roundedUp = "10.0", roundedDown = "0.0"),
+    TestCase(bigDecimal  = "15.000000001", roundedUp = "20.0", roundedDown = "10.0"),
+    TestCase(bigDecimal  = "10.000000001", roundedUp = "20.0", roundedDown = "10.0"),
+    TestCase(bigDecimal  = "100000000000010.000000001", roundedUp = "100000000000020.0", roundedDown = "100000000000010.0")
```
Comment (johnrevill) [inline test/ssttpcalculator/BigDecimalUtilTest.scala]: Always get a little nervous around rounding of money.
Some other possible paranoia boundary cases:
`bigDecimal = "9.9"`
`bigDecimal = "19.9"`
`bigDecimal = "0.0"`
`bigDecimal = "10.0"`
`bigDecimal = "15.5"`

**Review by johnrevill — APPROVED** [review]

File: app/ssttpcalculator/BigDecimalUtil.scala (hunk 1 above)
Comment (pawelpanasewicz) [reply app/ssttpcalculator/BigDecimalUtil.scala]: yeah, that is intended. We support values from 32 to 30000 and want to display a number between divided by  another number from range 3 to 12. 


>Iirc from ancient Java days, BigDecimal has methods to support careful/exact rounding of numbers avoiding the need for using the long value there.

Yes, I know that, but these methods sets the precision inside the big decimal. I didn't want to create a decimal with such precision and returning it back, because it was not the purpose of this method. 
I could set the precision and rounding mode and then reset them to back what it was, but decided to follow existing convention.

File: test/ssttpcalculator/BigDecimalUtilTest.scala (hunk 3 above)
Comment (pawelpanasewicz) [reply test/ssttpcalculator/BigDecimalUtilTest.scala]: i'll add more cases

File: test/pagespecs/PaymentSummaryPageSpec.scala (hunk 2 above)
Comment (pawelpanasewicz) [reply test/pagespecs/PaymentSummaryPageSpec.scala]: yes, any clue how to change it?

File: app/ssttpcalculator/BigDecimalUtil.scala [hunk 4]
```diff
@@ -18,8 +18,14 @@ package ssttpcalculator
...
 
-  def roundDownToNearestTen(value: BigDecimal): BigDecimal = BigDecimal((value.longValue() / 10) * 10)
+  def roundDownToNearestTen(value: BigDecimal): BigDecimal = {
+    BigDecimal((value.longValue() / 10) * 10)
+  }
 
-  def roundUpToNearestTen(value: BigDecimal): BigDecimal = BigDecimal((value.longValue() / 10) * 10) + 10
+  def roundUpToNearestTen(value: BigDecimal): BigDecimal = {
+    val roundedLong = (value.longValue() / 10) * 10
+    if (roundedLong == value) value //when no rounding is needed (value is exact multiplication of 10)
```
Comment (johnrevill) [inline app/ssttpcalculator/BigDecimalUtil.scala:27]: Oh yeah, I see it now :)

File: app/ssttpcalculator/BigDecimalUtil.scala (hunk 1 above)
Comment (pawelpanasewicz) [reply app/ssttpcalculator/BigDecimalUtil.scala]: ps. looks like bigdecimail doesn't support rounding to 10 via `setScale` and `roundingMode` thats why so hacky solution
